.github/workflows/ci.yml

```diff
@@ -0,0 +1,48 @@
+name: ci
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  ci:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            target-folder: drop-linux
+          - os: windows-latest
+            target-folder: drop
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "20"
+
+      - name: npm ci
+        run: npm ci
+
+      - name: lint
+        run: npm run lint
+
+      - name: prettier
+        run: npm run prettier
+
+      - name: test
+        run: npm test
+
+      - name: pack
+        run: npm pack
+
+      - name: Copy Files to target-folder
+        run: mkdir -p ${{ matrix.target-folder }} && cp *.tgz ${{ matrix.target-folder }}
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.target-folder }}
+          path: ${{ matrix.target-folder }}
```

Comment:
What is the benefit of doing `npm pack` and the remaining steps?

I think they [may have made sense](https://github.com/Azure/openapi-diff/blob/main/eng/test-steps.yml) in the context of uploading tests results to Azure DevOps pipeline and/or publishing the package, but if we make this GH action go up to `test` and no later, will uploading artifacts help? Will it somehow integrate with GitHub UI for nice test viewing experience or similar? If not, perhaps we can remove it.

Reply:
Yes, it does!

Example here in a repo I also maintain: https://github.com/gitify-app/gitify/actions/runs/9626236252

You can see all of the artifacts of a given workflow, which is something that I'd say would be nice to keep!